src/utils/bond.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <cstddef>
+#include <memory>
+#include "storage/v2/delta.hpp"
+#include "utils/memory.hpp"
+#include "utils/pmr/list.hpp"
+
+template <typename Container>
+struct Bond {
+  explicit Bond(std::size_t initial_size)
+      : res_(std::make_unique<resource>(initial_size)),
+        container_(memgraph::utils::Allocator<Container>(res_.get()).template new_object<Container>()){};
+
+  Bond(Bond &&other) noexcept : res_(std::move(other.res_)), container_(other.container_) {
+    other.container_ = nullptr;
+    [[maybe_unused]] auto _ = other.res_.release();
+  }
```

Comment:
useful utility std::exchange
```suggestion
  Bond(Bond &&other) noexcept : res_(std::exchange(other.res_, nullptr)), container_(other.container_) {
    other.container_ = nullptr;
  }
```

Reply:
Nice!